### ISSUE TITLE
Change content at specific TAG before build

### ISSUE THREAD

**Issue by chingling123** [ZhgChgLi/ZMarkupParser#23]
I need to change the text at a specific tag, like H2 I need that all H2 tag needs to but uppercase.

**Comment by zhgchgli0718** [comment]
Let me confirm first, are you requesting to convert the text to uppercase or lowercase? If so, I can consider capitalization as a style feature and extend it to MarkupStyle. This is because capitalization requires modifying the actual text itself, which is not controlled by NSAttributedString Attributes.

**Comment by chingling123** [comment]
Yes, I need to change to uppercase the content of the H2 tag for example…

Sent from my iPhone

On 8 Jul 2023, at 03:57, ZhgChgLi ***@***.***> wrote:

﻿

Let me confirm first, are you requesting to convert the text to uppercase or lowercase? If so, I can consider capitalization as a style feature and extend it to MarkupStyle. This is because capitalization requires modifying the actual text itself, which is not controlled by NSAttributedString Attributes.

**Comment by chingling123** [comment]
Thanks, it's working, I'm using the main branch from swift package.
BTW, when I have this html: <h2 style="text-align: justify;"><span style="color: #e67e23;"><span style="color: #000000;">heading 2</span></span></h2>

the H2 custom style not working at all, I can change the color, but font and the new font case is ignored, it's something that I need to do at the code?

![H2](https://github.com/ZhgChgLi/ZMarkupParser/assets/5823476/267125d4-912a-4ac9-9623-8090be3531c2)

**Comment by zhgchgli0718** [comment]
> Thanks, it's working, I'm using the main branch from swift package. BTW, when I have this html:
> 
> ## heading 2
> the H2 custom style not working at all, I can change the color, but font and the new font case is ignored, it's something that I need to do at the code?
> 
> ![H2](https://user-images.githubusercontent.com/5823476/252723266-267125d4-912a-4ac9-9623-8090be3531c2.jpg)

could you provide the full code?
I've try:
```
<h2 style="text-align: justify;"><span style="color: #e67e23;"><span style="color: #0000ff;">heading 2</span></span></h2>
```
![image](https://github.com/ZhgChgLi/ZMarkupParser/assets/33706588/911bf33e-ddae-4e2e-b83f-580dc7fbe9f7)
it's working, h2 color has set to 0000ff

**Comment by chingling123** [comment]
Hello,

Yes, but the font size not change.


Swift:

 self.parser = parser.add(H2_HTMLTagName(), withCustomStyle: MarkupStyle(font:MarkupStyleFont(UIFont.bold(14)), fontCase: MarkupStyleFont.CaseStyle.uppercase, foregroundColor:MarkupStyleColor.init(color: .unilabsYellow)))

HTML:


<p>Texto de teste de uma not&iacute;cia</p>
<p><strong>bold&nbsp;</strong><em>it&aacute;lico&nbsp;</em><span style="text-decoration: underline;">sublinhado</span> <span style="text-decoration: line-through;">rasurado</span> <span style="color: #e67e23;">com cor<span style="color: #000000;"> <a href="https://www.google.com">com link na mesma p&aacute;gina</a> <a href="https://www.google.com" target="_blank" rel="noopener">com link numa nova p&aacute;gina</a></span></span></p>
<p style="text-align: left;"><span style="color: #e67e23;"><span style="color: #000000;">texto alinhado &agrave; esquerda texto alinhado &agrave; esquerda texto alinhado &agrave; esquerda</span></span></p>
<p style="text-align: center;"><span style="color: #e67e23;"><span style="color: #000000;">texto centrado texto centrado texto centrado texto centrado texto centrado texto centrado</span></span></p>
<p style="text-align: right;"><span style="color: #e67e23;"><span style="color: #000000;">texto alinhado &agrave; direita</span></span></p>
<p style="text-align: justify;"><span style="color: #e67e23;"><span style="color: #000000;">texto justificado texto justificado texto justificado texto justificado texto justificado texto justificado</span></span></p>
<h2 style="text-align: justify;"><span style="color: #e67e23;"><span style="color: #000000;">heading 2</span></span></h2>
<h3><span style="color: #e67e23;"><span style="color: #000000;">heading 3</span></span></h3>
<h4><span style="color: #e67e23;"><span style="color: #000000;">heading 4</span></span></h4>
<h5><span style="color: #e67e23;"><span style="color: #000000;">heading 5</span></span></h5>
<p class="text-list"><span style="color: #e67e23;"><span style="color: #000000;">text list</span></span></p>
<p class="text-list text-small"><span style="color: #e67e23;"><span style="color: #000000;">text small</span></span></p>
<ul>
<li class="text-list text-small"><span style="color: #e67e23;"><span style="color: #000000;">item 1</span></span></li>
<li class="text-list text-small"><span style="color: #e67e23;"><span style="color: #000000;">item 2</span></span></li>
<li class="text-list text-small">
<h3><span style="color: #e67e23;"><span style="color: #000000;">item 3</span></span></h3>
</li>
<li class="text-list text-small">
<h4><span style="color: #e67e23;"><span style="color: #000000;">item 4</span></span></h4>
</li>
<li class="text-list text-small">
<h5><span style="color: #e67e23;"><span style="color: #000000;">item 5</span></span></h5>
</li>
</ul>
<ol>
<li><span style="color: #e67e23;"><span style="color: #000000;">ponto 1</span></span></li>
<li><span style="color: #e67e23;"><span style="color: #000000;">ponto 2</span></span></li>
<li><span style="color: #e67e23;"><span style="color: #000000;">ponto 3</span></span></li>
</ol>
<table style="border-collapse: collapse; width: 100%;" border="1">
<tbody>
<tr>
<td style="width: 22.481%;">tabela 1</td>
<td style="width: 22.481%;">1.1</td>
<td style="width: 22.481%;">1.2</td>
<td style="width: 22.481%;">1.3</td>
</tr>
<tr>
<td style="width: 22.481%;">tabela 2</td>
<td style="width: 22.481%;">2.1</td>
<td style="width: 22.481%;">2.2</td>
<td style="width: 22.481%;">2.3</td>
</tr>
<tr>
<td style="width: 22.481%;">tabela 3</td>
<td style="width: 22.481%;">3.1</td>
<td style="width: 22.481%;">3.2</td>
<td style="width: 22.481%;">3.3</td>
</tr>
</tbody>
</table>


Thanks

Erik

On 12 Jul 2023, at 14:50, ZhgChgLi ***@***.***> wrote:



Thanks, it's working, I'm using the main branch from swift package. BTW, when I have this html:

heading 2

the H2 custom style not working at all, I can change the color, but font and the new font case is ignored, it's something that I need to do at the code?

[H2]<https://user-images.githubusercontent.com/5823476/252723266-267125d4-912a-4ac9-9623-8090be3531c2.jpg>

could you provide the full code?
I've try:

<h2 style="text-align: justify;"><span style="color: #e67e23;"><span style="color: #0000ff;">heading 2</span></span></h2>


[image]<https://user-images.githubusercontent.com/33706588/252992963-911bf33e-ddae-4e2e-b83f-580dc7fbe9f7.png>
it's working, h2 color has set to 0000ff

**Comment by zhgchgli0718** [comment]
Hello, I tested the following code:
```swift
let parser = ZHTMLParserBuilder.initWithDefault().add(H2_HTMLTagName(), withCustomStyle: MarkupStyle(font:MarkupStyleFont(UIFont.boldSystemFont(ofSize: 14)), fontCase: MarkupStyleFont.CaseStyle.uppercase, foregroundColor:MarkupStyleColor.init(color: .yellow))).build()

let textView = UITextView()
textView.frame.size.width = 390
textView.isScrollEnabled = false
textView.backgroundColor = .white
textView.setHtmlString("sss <h2 style=\"text-align: justify;\"><span style=\"color: #e67e23;\"><span style=\"color: #0000ff;\">heading 2</span></span></h2> sss", with: parser)
textView.layoutIfNeeded()
```
![image](https://github.com/ZhgChgLi/ZMarkupParser/assets/33706588/dfc90710-9856-4d2e-bbe3-19a749268dbd)

The result is normal.

I suspect that your use of `UIFont.bold(14)` as a non-official Foundation UIFont initialization might be causing the issue.

Can you try `print(UIFont.bold(14).pointSize)`?

Alternatively, you can directly declare a custom bold font size using MarkupStyleFont(size: 15, weight: .bold).

Thank you!

**Comment by chingling123** [comment]
I’ll try, but the others, H3..H5, I’m using the same custom style


On 12 Jul 2023, at 15:21, ZhgChgLi ***@***.***> wrote:



Hello, I tested the following code:

let parser = ZHTMLParserBuilder.initWithDefault().add(H2_HTMLTagName(), withCustomStyle: MarkupStyle(font:MarkupStyleFont(UIFont.boldSystemFont(ofSize: 14)), fontCase: MarkupStyleFont.CaseStyle.uppercase, foregroundColor:MarkupStyleColor.init(color: .yellow))).build()

let textView = UITextView()
textView.frame.size.width = 390
textView.isScrollEnabled = false
textView.backgroundColor = .white
textView.setHtmlString("sss <h2 style=\"text-align: justify;\"><span style=\"color: #e67e23;\"><span style=\"color: #0000ff;\">heading 2</span></span></h2> sss", with: parser)
textView.layoutIfNeeded()

[image]<https://user-images.githubusercontent.com/33706588/253001737-dfc90710-9856-4d2e-bbe3-19a749268dbd.png>

The result is normal.

I suspect that your use of UIFont.bold(14) as a non-official Foundation UIFont initialization might be causing the issue.

Can you try print(UIFont.bold(14).pointSize)?

Alternatively, you can directly declare a custom bold font size using MarkupStyleFont(size: 15, weight: .bold).

Thank you!

**Comment by chingling123** [comment]
Printed: FONT: 14.0

**Comment by zhgchgli0718** [comment]
Should be fixed in the latest release (v1.4.0). Thank you for your feedback.
